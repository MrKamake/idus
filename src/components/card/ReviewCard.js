import React from 'react';
import './ReviewCard.scss';
import Rating from 'react-rating';
import { TiStarOutline, TiStarFullOutline } from 'react-icons/ti';

const ReviewCard = ({ reviewer, title, coverImage, rating, comment }) => {
  return (
    <div className="ui_card--side">
      <div className="ui_card__imgcover">
        <img className="ui_card__img" src={coverImage} alt="cover" />
      </div>
      <div className="ui_card__txtarea">
        <div className="ui_card__info">
          <span className="ui_card__title">{title}</span>
          {comment && <span className="ui_card__comment">{comment}</span>}
        </div>
        {rating && (
          <div className="ui_card__rating">
            <div className="ui_rating_wrapper">
              <Rating
                className="ui_rating"
                readonly
                initialRating={rating}
                emptySymbol={<TiStarOutline size="20px" />}
                fullSymbol={<TiStarFullOutline size="20px" />}
              />
              <span className="ui_card__reviewer"> | {reviewer}</span>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

export default ReviewCard;
