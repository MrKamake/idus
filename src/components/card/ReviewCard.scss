@import '../../styles/utils.scss';

.ui_card--side {
  position: relative;
  display: table;
  @include cardForm;
  * {
    display: table-cell;
    vertical-align: top;
  }
  .ui_card__imgcover {
    overflow: hidden;
    .ui_card__img {
      width: 170px;
    }
  }
  .ui_card__txtarea {
    position: relative;
    padding-bottom: 45px;
    .ui_card__info {
      width: 230px;
      height: auto;
      padding: 0 20px;
      .ui_card__title {
        display: block;
        display: -webkit-box;
        margin: 15px 0;
        height: 21px;
        font: {
          size: 14px;
          weight: 700;
        }
        line-height: 1.5;
        -webkit-line-clamp: 1;
        -webkit-box-orient: vertical;
        overflow: hidden;
        text-overflow: ellipsis;
      }
      .ui_card__comment {
        color: $gray6;
        display: -webkit-box;
        height: 74px;
        line-height: 1.5;
        -webkit-line-clamp: 5;
        -webkit-box-orient: vertical;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: pre-line;
        font-size: 14px;
      }
    }
  }
  .ui_card__rating {
    position: absolute;
    bottom: 0;
    padding: 14px 20px;
    width: 100%;
    height: auto;
    border: 0;
    box-sizing: border-box;
    .ui_rating_wrapper {
      .ui_rating {
        display: inline-block;
        color: $gold;
      }
      .ui_card__reviewer {
        display: inline-block;
        vertical-align: sub;
        font-size: 13px;
        color: $gray3;
      }
    }
  }
}
