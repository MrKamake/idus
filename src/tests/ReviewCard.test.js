import React from 'react';
import { shallow } from 'enzyme';
import ReviewCard from '../components/card/ReviewCard';

describe('ReviewCard', () => {
  describe('<ReviewCard />', () => {
    let wrapper, props;

    beforeEach(() => {
      props = {
        title: '상품 후기',
        created_by: 'John Doe',
        cover_image: '목걸이',
        rating: 3,
        comment: '목걸이가 반짝거리고 고급스럽게 이뻐요~'
      };
      wrapper = shallow(<ReviewCard {...props} />);
    });

    it('should contain elements', () => {
      expect(
        wrapper.contains(
          <img className="ui_card__img" src="목걸이" alt="cover" />
        )
      ).toBe(true);
      expect(
        wrapper.contains(<span className="ui_card__reviewer"> | John Doe</span>)
      ).toBe(true);
      expect(
        wrapper.contains(<span className="ui_card__title">상품 후기</span>)
      ).toBe(true);
      expect(wrapper.find('.ui_rating').length).toBe(1);
      expect(wrapper.find('.ui_card__comment').length).toBe(1);
      expect(
        wrapper.contains(
          <span className="ui_card__comment">
            목걸이가 반짝거리고 고급스럽게 이뻐요~
          </span>
        )
      ).toBe(true);
    });
  });
});
