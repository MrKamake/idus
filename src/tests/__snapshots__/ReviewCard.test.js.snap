// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCard <ReviewCard /> matches snapshot 1`] = `
<div
  className="ui_card--side"
>
  <div
    className="ui_card__imgcover"
  >
    <t
      afterLoad={[Function]}
      alt="cover"
      beforeLoad={[Function]}
      className="ui_card__img"
      delayMethod="throttle"
      delayTime={300}
      effect="blur"
      placeholderSrc={null}
      src="목걸이"
      threshold={100}
      useIntersectionObserver={true}
      visibleByDefault={false}
      wrapperClassName=""
    />
  </div>
  <div
    className="ui_card__txtarea"
  >
    <div
      className="ui_card__info"
    >
      <span
        className="ui_card__title"
      >
        상품 후기
      </span>
      <span
        className="ui_card__comment"
      >
        목걸이가 반짝거리고 고급스럽게 이뻐요~
      </span>
    </div>
    <div
      className="ui_card__rating"
    >
      <div
        className="ui_rating_wrapper"
      >
        <RatingAPILayer
          className="ui_rating"
          direction="ltr"
          emptySymbol={
            <TiStarOutline
              size="20px"
            />
          }
          fractions={1}
          fullSymbol={
            <TiStarFullOutline
              size="20px"
            />
          }
          initialRating={3}
          onChange={[Function]}
          onClick={[Function]}
          onHover={[Function]}
          placeholderSymbol={
            Object {
              "backgroundColor": "red",
              "border": "5px double white",
              "borderRadius": "50%",
              "display": "inline-block",
              "height": 30,
              "width": 30,
            }
          }
          quiet={false}
          readonly={true}
          start={0}
          step={1}
          stop={5}
        />
        <span
          className="ui_card__reviewer"
        >
           | 
          John Doe
        </span>
      </div>
    </div>
  </div>
</div>
`;
